distribution/push_v2.go

```diff
@@ -670,10 +668,9 @@ func numOfMatchingPathComponents(pth string, matchComponents []string) int {
 }
 
 func getPathComponents(path string) []string {
-	// make sure to add docker.io/ prefix to the path
-	named, err := reference.ParseNamed(path)
+	named, err := reference.ParseNormalizedNamed(path)
```

Comment:
I don't understand why we're reparsing here. The function is passed `repoInfo.Name()` Couldn't it just take `repoInfo` as an argument and return `strings.Split(repoInfo.Name(), "/")`?